src/main/java/uk/gov/hmcts/reform/divorce/orchestration/service/impl/CaseServiceImpl.java

```diff
@@ -43,18 +73,53 @@
     }
 
     @Override
-    public CaseDataResponse getCase(final String authorizationToken) {
-        final CaseDetails caseDetails = cmsClient.getCaseFromCcd(authorizationToken);
+    public GetCaseResponse getCase(final String authorizationToken) throws CaseNotFoundException {
+        User user = getUser(authorizationToken);
 
-        log.info("Successfully retrieved case with id {} and state {}", caseDetails.getCaseId(), caseDetails.getState());
+        List<CaseDetails> caseDetailsList = getCaseListForUser(user);
 
-        final Map<String, Object> caseData = caseDetails.getCaseData();
+        if (isEmpty(caseDetailsList)) {
+            throw new CaseNotFoundException("No case found for user id " + user.getUserDetails().getId());
+        }
 
-        return CaseDataResponse.builder()
-            .caseId(caseDetails.getCaseId())
+        if (caseDetailsList.size() > 1) {
+            throw new DuplicateCaseException(String.format("There are [%d] cases for the user [%s]",
+                caseDetailsList.size(), user.getUserDetails().getId()));
+        }
+
+        CaseDetails caseDetails = caseDetailsList.get(0);
+
+        log.info("Successfully retrieved case for user id {} with case id {} and state {}",
+            user.getUserDetails().getId(),
+            caseDetails.getId(),
+            caseDetails.getState()
+        );
+
+        return GetCaseResponse.builder()
+            .id(String.valueOf(caseDetails.getId()))
             .state(caseDetails.getState())
-            .court(String.valueOf(caseData.get(D_8_DIVORCE_UNIT)))
-            .data(caseData)
+            .data(caseDetails.getData())
             .build();
     }
+
+    private List<CaseDetails> getCaseListForUser(User user) {
+        return Optional.ofNullable(
+            coreCaseDataApi.searchForCitizen(
+                user.getAuthToken(),
+                authTokenGenerator.generate(),
+                user.getUserDetails().getId(),
```

Comment:
doesn't that do an API call to idam?

Reply:
actually call is made here so in above we are just accessing object.
```
private User getUser(String authorisation) {
        final String bearerToken = authUtil.getBearerToken(authorisation);
        final UserDetails userDetails = idamClient.getUserDetails(bearerToken);

        log.info("Successfully retrieved user id from Idam {}", userDetails.getId());

        return new User(bearerToken, userDetails);
    }
```